Lib/test/libregrtest/main.py

```diff
@@ -129,14 +129,19 @@ def __init__(self, ns: Namespace, _add_python_opts: bool = False):
 
         # Randomize
         self.randomize: bool = ns.randomize
-        self.random_seed: int | None =  (
-            ns.random_seed
-            if ns.random_seed is not None
-            else random.getrandbits(32)
-        )
-        if 'SOURCE_DATE_EPOCH' in os.environ:
+        if ('SOURCE_DATE_EPOCH' in os.environ
+            # don't use the variable if empty
+            and os.environ['SOURCE_DATE_EPOCH']
+        ):
             self.randomize = False
-            self.random_seed = None
+            # SOURCE_DATE_EPOCH should be an integer, but use a string to not
+            # fail if it's not integer. random.seed() accepts a string.
+            # https://reproducible-builds.org/docs/source-date-epoch/
+            self.random_seed: int | str = os.environ['SOURCE_DATE_EPOCH']
+        elif ns.random_seed is None:
+            self.random_seed = random.getrandbits(32)
```

Comment:
If so, it would become:

```suggestion
            self.random_seed = str(random.getrandbits(32))
```